### PR TITLE
Issue #3445902: Apply static caching to the SocialPrivateMessageService

### PR DESCRIPTION
**Pull request by bramtenhove**
## Problem
The `updateUnreadCount()` function in `SocialPrivateMessageService` is sometimes called multiple times during a request.

## Solution
Apply simple static caching.

## Issue tracker
https://www.drupal.org/project/social/issues/3445902

## Theme issue tracker
N/a

## How to test
- [ ] Send a private message to a user
- [ ] Observe their unread counter remains functional

## Screenshots
N/a

## Release notes
Minor performance improvement for the private message feature.

## Change Record
N/a

## Translations
N/a

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

File: modules/social_features/social_private_message/src/Service/SocialPrivateMessageService.php [hunk 1]
```diff
@@ -63,41 +63,46 @@ public function deleteUserDataThreadInfo(EntityBase $entity) {
...
-    $unread = 0;
+    $unread = &drupal_static(__FUNCTION__);
 
-    // Get the user.
-    $uid = $this->currentUser->id();
-    /** @var \Drupal\user\UserStorageInterface $user_storage */
-    $user_storage = $this->userManager;
-    /** @var \Drupal\user\UserInterface $user */
-    $user = $user_storage->load($uid);
+    if (!isset($unread)) {
```
Comment (bramtenhove) [inline modules/social_features/social_private_message/src/Service/SocialPrivateMessageService.php:68]: This is the only actual change together with line `66`, the rest is indentation.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Review by ribel — APPROVED** [review]
Changes looks good to me, also tested and there are no regressions. 
image

image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3893-vetsc95sdim4tgdllmf50zjrrs15pjzq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/663b1efabf7868a563f57d17

**Comment by bramtenhove** [comment]
See f4675e79acc7a54fbe2ad09b5d7d4f8ce827cabe (12.3.x) and 1d9f048cc79a561b3efd30a60cdbe429e2b5ea0b (12.4.x)

**Pull request merged**